{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "pth_to_stats = '/home/yulia/parampampam/raw_data/stats/18S_and_ITS/stats_final.tsv'\n",
    "pth_to_ssu = '/home/yulia/parampampam/raw_data/fasta_files/18S/formated_id_full.fa'\n",
    "pth_to_its = '/home/yulia/parampampam/raw_data/fasta_files/ITS/formated_id_with_add.fa'\n",
    "\n",
    "outf_ssu = '/home/yulia/parampampam/raw_data/fasta_files/18S_and_ITS/18S.fa'\n",
    "outf_its = '/home/yulia/parampampam/raw_data/fasta_files/18S_and_ITS/ITS.fa'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "47\n",
      "954787BF1B\n",
      "55CEC58B7B\n",
      "A22A7EB077\n",
      "84CE765D07\n",
      "A9D77FB142\n",
      "C0FF30338F\n",
      "6A22C8816B\n",
      "A100F8A811\n",
      "FFCD65A9C1\n",
      "FE6BF56E99\n",
      "BAFBB32B3C\n",
      "9F17B676E6\n",
      "A28E935B8D\n",
      "E4C28EC0EC\n",
      "D4CE20BD5A\n",
      "E24B3A4289\n",
      "7EB0465780\n",
      "87876A753A\n",
      "5EDFE55A10\n",
      "3B4F40C1B6\n",
      "4C5F4E4CDE\n",
      "A50EB0413A\n",
      "20D1316E5D\n",
      "8E203F786B\n",
      "4571A227E6\n",
      "317A91CB34\n",
      "754C622D24\n",
      "AFED3029BB\n",
      "0C044FF14F\n",
      "F5F4A73DA3\n",
      "3232DFC4C6\n",
      "70890594D2\n",
      "7AB785E8EB\n",
      "F2145CCC2C\n",
      "7760F61162\n",
      "ABB2EFBA0D\n",
      "06BCC826BF\n",
      "7337CC22D1\n",
      "55622BCCF2\n",
      "E5A80E2658\n",
      "1B54F3DD84\n",
      "4EC675C343\n",
      "05EB3CB2EF\n",
      "FT8_ITS\n",
      "E53453BD54\n",
      "E53453BD54 here\n",
      "27457DB888\n",
      "27457DB888 here\n",
      "B9FA7E99CC\n",
      "B9FA7E99CC here\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "from Bio import SeqIO\n",
    "\n",
    "ssu_out = []\n",
    "its_out = []\n",
    "\n",
    "\n",
    "df = pd.read_csv(pth_to_stats, sep='\\t')\n",
    "ssu_recs = [rec for rec in SeqIO.parse(pth_to_ssu, 'fasta')]\n",
    "its_recs = [rec for rec in SeqIO.parse(pth_to_its, 'fasta')]\n",
    "\n",
    "foursev_its_recs = []\n",
    "c = 0\n",
    "with open(outf_its, 'w') as out_f:\n",
    "    for its in df.its_id:\n",
    "        foursev_its_recs.append(its)\n",
    "        c+=1\n",
    "        for rec in its_recs:\n",
    "            if its == rec.name:\n",
    "                its_out.append(rec)\n",
    "                SeqIO.write(rec, out_f, 'fasta')\n",
    "print(c)\n",
    "\n",
    "foursev_ssu_recs = []\n",
    "c = 0\n",
    "with open(outf_ssu, 'w') as out_f:\n",
    "    for ssu in df.ssu_id:\n",
    "        foursev_ssu_recs.append(ssu)\n",
    "        c+=1\n",
    "        for rec in ssu_recs:\n",
    "            if ssu == rec.name:\n",
    "                ssu_out.append(rec)\n",
    "                SeqIO.write(rec, out_f, 'fasta')\n",
    "\n",
    "ssu_names = set()\n",
    "with open(outf_ssu, 'r') as ssu_f:\n",
    "    for line in ssu_f:\n",
    "        line = line.strip()\n",
    "        if line.startswith('>'):\n",
    "            ssu_names.add(line[1:])\n",
    "            \n",
    "its_names = set()\n",
    "with open(outf_its, 'r') as its_f:\n",
    "    for line in its_f:\n",
    "        line = line.strip()\n",
    "        if line.startswith('>'):\n",
    "            its_names.add(line[1:])\n",
    "\n",
    "# check fails\n",
    "for name in foursev_its_recs:\n",
    "    print(name)\n",
    "    if name not in its_names:\n",
    "        print(name, 'here')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [],
   "source": [
    "# для Максима\n",
    "import pandas as pd\n",
    "\n",
    "def rename_ids(stats, in_f, out_f):\n",
    "    df = pd.read_csv(stats, sep='\\t')\n",
    "    with open(in_f) as inf, open(out_f, 'w') as outf:\n",
    "        for line in inf:\n",
    "            for idx, descr in zip(df['Unnamed: 0'], df['description']):\n",
    "                if idx in line:\n",
    "                    line = line.replace(idx, descr)\n",
    "            outf.write(line)\n",
    "                \n",
    "                \n",
    "stats = '/home/yulia/parampampam/raw_data/stats/COI/stats_final.tsv'\n",
    "in_f = '/home/yulia/parampampam/alignments/COI/tree_4_more_outgroup/mafft_aln_fianl_more_outgroup_full.fa'\n",
    "out_f = '/home/yulia/parampampam/alignments/COI/tree_4_more_outgroup/COI_normal_names.fa'\n",
    "\n",
    "stats = '/home/yulia/parampampam/raw_data/stats/18S/total_stats_full.tsv'\n",
    "in_f = '/home/yulia/parampampam/alignments/18S/18S_GBlocks_regions_3.fst'\n",
    "out_f = '/home/yulia/parampampam/alignments/18S/SSU_normal_names.fa'\n",
    "\n",
    "stats = '/home/yulia/parampampam/raw_data/stats/ITS/total_stats_full.tsv'\n",
    "in_f = '/home/yulia/parampampam/alignments/ITS/ITS_GBlocks_2.fst'\n",
    "out_f = '/home/yulia/parampampam/alignments/ITS/ITS_normal_names.fa'\n",
    "\n",
    "rename_ids(stats, in_f, out_f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "outputhat23=16\n",
      "treein = 0\n",
      "compacttree = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "All-to-all alignment.\n",
      "tbfast-pair (nuc) Version 7.429\n",
      "alg=L, model=DNA200 (2), 2.00 (6.00), -0.10 (-0.30), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "outputhat23=16\n",
      "Loading 'hat3.seed' ... \n",
      "done.\n",
      "Writing hat3 for iterative refinement\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "tbutree = 1, compacttree = 0\n",
      "Constructing a UPGMA tree ... \n",
      "   40 / 47\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "STEP    45 /46 \n",
      "Reallocating..done. *alloclen = 6654\n",
      "STEP    46 /46 \n",
      "done.\n",
      "tbfast (nuc) Version 7.429\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "1 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 16\n",
      "sueff_global = 0.100000\n",
      "nadd = 16\n",
      "Loading 'hat3' ... done.\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "   40 / 47\n",
      "Segment   1/  1    1-2851\n",
      "STEP 003-006-1  identical.    identical.    accepted. identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    rejected. identical.    identical.    identical.    rejected. rejected. identical.    identical.    identical.    rejected. identical.    rejected. identical.    rejected. identical.    accepted. identical.    identical.    identical.    identical.    identical.    rejected. rejected. identical.    identical.    identical.    identical.    rejected. identical.    identical.    accepted. accepted. accepted. identical.    identical.    accepted. identical.    identical.    identical.    accepted. identical.    identical.    accepted. identical.    identical.    rejected. identical.    identical.    rejected. identical.    rejected. identical.    identical.    identical.    identical.    identical.    identical.    rejected. rejected. identical.    rejected. identical.    identical.    rejected. identical.    rejected. identical.    identical.    identical.    identical.    rejected. identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.   \n",
      "Converged.\n",
      "\n",
      "done\n",
      "dvtditr (nuc) Version 7.429\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<16) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "CPU times: user 1.4 s, sys: 224 ms, total: 1.62 s\n",
      "Wall time: 51.3 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "! mafft --maxiterate 1000 --localpair /home/yulia/parampampam/raw_data/fasta_files/18S_and_ITS/ITS.fa > /home/yulia/parampampam/raw_data/fasta_files/18S_and_ITS/ITS_aln.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "outputhat23=16\n",
      "treein = 0\n",
      "compacttree = 0\n",
      "stacksize: 8192 kb\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "All-to-all alignment.\n",
      "tbfast-pair (nuc) Version 7.429\n",
      "alg=L, model=DNA200 (2), 2.00 (6.00), -0.10 (-0.30), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "outputhat23=16\n",
      "Loading 'hat3.seed' ... \n",
      "done.\n",
      "Writing hat3 for iterative refinement\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "Gap Penalty = -1.53, +0.00, +0.00\n",
      "tbutree = 1, compacttree = 0\n",
      "Constructing a UPGMA tree ... \n",
      "   40 / 47\n",
      "done.\n",
      "\n",
      "Progressive alignment ... \n",
      "STEP    41 /46 \n",
      "Reallocating..done. *alloclen = 6844\n",
      "STEP    46 /46 \n",
      "done.\n",
      "tbfast (nuc) Version 7.429\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "1 thread(s)\n",
      "\n",
      "minimumweight = 0.000010\n",
      "autosubalignment = 0.000000\n",
      "nthread = 0\n",
      "randomseed = 0\n",
      "blosum 62 / kimura 200\n",
      "poffset = 0\n",
      "niter = 16\n",
      "sueff_global = 0.100000\n",
      "nadd = 16\n",
      "Loading 'hat3' ... done.\n",
      "generating a scoring matrix for nucleotide (dist=200) ... done\n",
      "\n",
      "   40 / 47\n",
      "Segment   1/  1    1-3086\n",
      "STEP 003-004-1  identical.    identical.    identical.    identical.    identical.    identical.    identical.    accepted. identical.    identical.    identical.    identical.    identical.    rejected. identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    rejected. identical.    identical.    rejected. identical.    rejected. identical.    identical.    identical.    rejected. identical.    identical.    identical.    identical.    rejected. identical.    rejected. identical.    rejected. identical.    identical.    accepted. rejected. identical.    accepted. accepted. accepted. identical.    accepted. identical.    accepted. accepted. identical.    rejected. rejected. accepted. rejected. identical.    rejected. rejected. rejected. identical.    identical.    rejected. rejected. identical.    rejected. identical.    rejected. identical.    identical.    identical.    identical.    rejected. identical.    identical.    identical.    identical.    identical.    rejected. identical.    identical.    identical.    rejected. identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    rejected. identical.    identical.    identical.    identical.    identical.    identical.    rejected. identical.    identical.    identical.    identical.    identical.    rejected. identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.    identical.   \n",
      "Converged.\n",
      "\n",
      "done\n",
      "dvtditr (nuc) Version 7.429\n",
      "alg=A, model=DNA200 (2), 1.53 (4.59), -0.00 (-0.00), noshift, amax=0.0\n",
      "0 thread(s)\n",
      "\n",
      "\n",
      "Strategy:\n",
      " L-INS-i (Probably most accurate, very slow)\n",
      " Iterative refinement method (<16) with LOCAL pairwise alignment information\n",
      "\n",
      "If unsure which option to use, try 'mafft --auto input > output'.\n",
      "For more information, see 'mafft --help', 'mafft --man' and the mafft page.\n",
      "\n",
      "The default gap scoring scheme has been changed in version 7.110 (2013 Oct).\n",
      "It tends to insert more gaps into gap-rich regions than previous versions.\n",
      "To disable this change, add the --leavegappyregion option.\n",
      "\n",
      "CPU times: user 1.73 s, sys: 263 ms, total: 1.99 s\n",
      "Wall time: 1min 6s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "! mafft --maxiterate 1000 --localpair /home/yulia/parampampam/raw_data/fasta_files/18S_and_ITS/18S.fa > /home/yulia/parampampam/raw_data/fasta_files/18S_and_ITS/18S_aln.fa"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "metadata": {},
   "outputs": [],
   "source": [
    "# назвать ssu сиквенсы айдишниками its \n",
    "from Bio import SeqIO\n",
    "\n",
    "ssu_in = '/home/yulia/parampampam/alignments/18S_and_ITS/18S_aln_regions.fa'\n",
    "its_in = '/home/yulia/parampampam/alignments/18S_and_ITS/ITS_aln_regions.fa'\n",
    "ssu_out ='/home/yulia/parampampam/alignments/18S_and_ITS/18S_aln_regions_renamed.fa'\n",
    "\n",
    "names = []\n",
    "its_recs = [rec for rec in SeqIO.parse(its_in, 'fasta')]\n",
    "for rec in its_recs:\n",
    "    names.append(rec.name)\n",
    "\n",
    "out_recs = []\n",
    "ssu_recs = [rec for rec in SeqIO.parse(ssu_in, 'fasta')]\n",
    "for rec, its_name in zip(ssu_recs, names):\n",
    "    rec.id = its_name\n",
    "    rec.name = its_name\n",
    "    rec.description = its_name\n",
    "    out_recs.append(rec)\n",
    "    \n",
    "with open(ssu_out, 'w') as out_f:\n",
    "    for rec in out_recs:\n",
    "        SeqIO.write(rec, out_f, 'fasta')"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
